src/main/java/com/checkmarx/jenkins/tools/CheckmarxInstaller.java

```diff
@@ -159,23 +182,42 @@ public Platform call() throws IOException {
         }
     }
 
+    public static class MyAuthenticator extends Authenticator {
+        String username;
+        String password;
+
+        public MyAuthenticator(String username, String password) {
+            this.username = username;
+            this.password = password;
+        }
+
+        protected PasswordAuthentication getPasswordAuthentication() {
+            if (getRequestorType().equals(RequestorType.PROXY)) {
+                return new PasswordAuthentication(username, password.toCharArray());
+            }
+            return super.getPasswordAuthentication();
+        }
+    }
+
     private static class Downloader extends MasterToSlaveCallable<Void, IOException> {
         private static final long serialVersionUID = 1L;
 
         private final URL downloadUrl;
         private final FilePath output;
         private final FilePath executableFile;
+        private final String proxy;
 
-        Downloader(URL downloadUrl, FilePath output, FilePath executableFile) {
+        Downloader(URL downloadUrl, String proxy, FilePath output, FilePath executableFile) {
             this.downloadUrl = downloadUrl;
             this.output = output;
             this.executableFile = executableFile;
+            this.proxy = proxy;
         }
 
         @Override
         public Void call() throws IOException {
             final File downloadedFile = new File(output.getRemote());
-            FileUtils.copyURLToFile(downloadUrl, downloadedFile, 10000, 10000);
+            copyURLToFile(downloadUrl, proxy, "", downloadedFile, 10000, 10000);
```

Comment:
why you are not passing the proxy type? the use don't give you? if not, maybe there is a way in regex / split to understand what type it, and if not, remove this parameter from the method input

Reply:
removed.